Vagrantfile

```diff
@@ -44,12 +52,28 @@ def bootstrap(box)
   <<-SCRIPT
   #{install}
 
-  if ! [ -f /home/vagrant/#{archive} ]; then
-    response=$(wget -nv https://storage.googleapis.com/golang/#{archive})
+  if [ -n "#{archive}" ]; then
+    if ! [ -f /home/vagrant/#{archive} ]; then
+      response=$(wget -nv #{downloadURL}/#{archive})
+    fi
+    tar -C /usr/local -xzf #{archive}
+  else
+    mkdir -p /usr/local/go/bootstrap
+    if ! [ -f #{bootstrap} ]; then
+      wget -nv #{downloadURL}/#{bootstrap}
+    fi
+    tar -C /usr/local/go/bootstrap -xzf #{bootstrap}
+    bash -c "cd /usr/local/go/bootstrap/go/src && ./make.bash"
+
+    if ! [ -f #{source} ]; then
+      wget -nv #{downloadURL}/#{source}
+    fi
+    tar -C /usr/local -xzf #{source}
+    bash -c "cd /usr/local/go/src && GOROOT_BOOTSTRAP=/usr/local/go/bootstrap/go ./make.bash"
   fi
-  tar -C /usr/local -xzf #{archive}
 
-  echo '#{profile}' >> /home/vagrant/.profile
+  home=~vagrant
```

Comment:
If I remember correctly this is executed as root, so both `$HOME/.profile` and `~/.profile` will change root's profile. But `~vagrant/.profile` should work.